components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceOrganizationServiceImpl.java

```diff
@@ -0,0 +1,307 @@
+/*
+ *   Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *   WSO2 Inc. licenses this file to you under the Apache License,
+ *   Version 2.0 (the "License"); you may not use this file except
+ *   in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing,
+ *   software distributed under the License is distributed on an
+ *   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *   KIND, either express or implied.  See the License for the
+ *   specific language governing permissions and limitations
+ *   under the License.
+ *
+ */
+package org.wso2.carbon.device.mgt.jaxrs.service.impl;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.device.mgt.common.DeviceOrganizationMetadataHolder;
+import org.wso2.carbon.device.mgt.common.DeviceOrganizationException;
+import org.wso2.carbon.device.mgt.common.InvalidConfigurationException;
+import org.wso2.carbon.device.mgt.common.DeviceOrganizationVisEdge;
+import org.wso2.carbon.device.mgt.common.DeviceOrganizationVisNode;
+import org.wso2.carbon.device.mgt.core.service.DeviceOrganizationProviderService;
+import org.wso2.carbon.device.mgt.core.service.DeviceOrganizationProviderServiceImpl;
+import org.wso2.carbon.device.mgt.jaxrs.service.api.DeviceOrganizationService;
+
+import javax.validation.Valid;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+import javax.ws.rs.core.Response;
+import java.util.List;
+
+public class DeviceOrganizationServiceImpl implements DeviceOrganizationService {
+    private static final Log log = LogFactory.getLog(DeviceAgentServiceImpl.class);
+
+    @POST
+    @Path("/add")
```

Comment:
@n-jay I don't think we need the URL "/add" for adding a resource when creating a REST API. Shall we remove the /add postfix from the url

Reply:
Fixed 8d78d79ba9c34b67ee36c2085699b3cacdd7033a